src/twisted/application/internet.py

```diff
@@ -1029,13 +1058,20 @@ def __init__(self, endpoint, factory, retryPolicy=None, clock=None):
             this attribute will not be serialized, and the default value (the
             reactor) will be restored when deserialized.
         @type clock: L{IReactorTime}
+
+        @param onNewConnection: A single argument L{callable} that may return a
+            L{Deferred} or L{None}. It will be called with the L{protocol
+            <interfaces.IProtocol>} after a new connection is made. Should it
+            raise an exception or the L{Deferred} fail, the connection attempt
+            is treated as a failure.
+        @type onNewConnection: L{callable}
```

Comment:
Why is this "onNewConnection" and not "onConnectionMade" ?

What does "new" mean?

I think that the docstring should be explicit that it is called for each connection (including calling it multiple times) on reconnection after a timeout.... or maybe just call onConnectionMade and this should be straightforward :)